### PR TITLE
Added Components , Pages and Products

### PR DESCRIPTION
**Pull request by haarsh24**
1. Converted static e-comm site to react project.
2. Added components and pages.
3. Added API calls to fetch products data from the mock backend.
4. Added context and reducers for the same.

### CODE REVIEW

File: src/components/featuredProducts/featuredProducts.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import "../styles/main.css";
+
+const FeaturedProducts = (props)=>{
+   // console.log(props);
```
Comment (xii-vi) [inline src/components/featuredProducts/featuredProducts.jsx:4]: Remove Console.log

File: src/components/featuredProducts/featuredProducts.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                        <i className="fa fa-heart-o "></i>
+                    </div>
+                    <button className="btn btn-add-to-cart  mb-1">
+                        Add to Cart
+                    </button>
+
+                </div>
+            </div>
+
+        </div> */}
```
Comment (xii-vi) [inline src/components/featuredProducts/featuredProducts.jsx:91]: Remove Commented code if not needed

**Review by xii-vi — APPROVED** [review]
_GO THROUGH THE COMMENT ONCE._

Good Work with the API calls Helper Component.

File: src/components/featuredProducts/featuredProducts.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    </div>
+                    <button className="btn btn-add-to-cart  mb-1">
+                        Add to Cart
+                    </button>
+
+                </div>
+            </div>
+
+        </div>
+        {/* <div className="product-collection item ">
```
Comment (yashpatel1797) [inline src/components/featuredProducts/featuredProducts.jsx:30]: remove commented code.

File: src/components/navbar/navbar.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <>
+         <div className="header">
+        <div className="bars">
+            <h2> <a className="logo" href="./index.html">SneakerXclub</a> </h2>
+
+           
+        </div>
+        <input className="input-text input-search" placeholder="Search" />
+        <div className="nav-icon">
+            <a href="./pages/login.html" className="h3 p1"> Login</a>
```
Comment (yashpatel1797) [inline src/components/navbar/navbar.jsx:16]: use Link tag from react-router dom

File: src/pages/homepage/homepage.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { FeaturedProducts } from "../../components/featuredProducts/featuredProducts";
+import { Footer } from "../../components/footer/footer";
+import { HeroSection } from "../../components/heroSection/heroSection";
+import { Navbar } from "../../components/navbar/navbar";
+import { ProductByCategory } from "../../components/productByCategory/productByCategory";
+import "../../components/styles/main.css";
```
Comment (yashpatel1797) [inline src/pages/homepage/homepage.jsx:7]: create index file inside components folder and export all from there so you don't require to write this every time.

**Review by yashpatel1797 — APPROVED** [review]
1. Added a few suggestions. the folder structure is good. 
2. The import thing is not mandatory but implementing this will be helpful. 
3. to  avoid moving back in folders using ../../
refer For Create React App without using Craco section in this https://dev.to/theodorusclarence/clean-up-your-imports-using-absolute-imports-alias-on-react-apps-next-js-cra-and-react-vite-2160

**Pull request merged**